test/Infrastructure.IntegrationTest/AdminConsole/Migrations/FinalFlexibleCollectionsDataMigrationsTests.cs

```diff
@@ -0,0 +1,334 @@
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user kept the Admin type and lost the editAssignedCollections and deleteAssignedCollections
+        // permissions but kept the accessEventLogs permission
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(OrganizationUserType.Admin, migratedOrgUser.Type);
+        Assert.NotEqual(orgUser.Permissions, migratedOrgUser.Permissions);
+        Assert.NotNull(migratedOrgUser.Permissions);
+        Assert.Contains("accessEventLogs", orgUser.Permissions);
+        Assert.Contains("editAssignedCollections", orgUser.Permissions);
+        Assert.Contains("deleteAssignedCollections", orgUser.Permissions);
+
+        Assert.Contains("accessEventLogs", migratedOrgUser.Permissions);
+        Assert.True(migratedOrgUser.GetPermissions().AccessEventLogs);
+        Assert.DoesNotContain("editAssignedCollections", migratedOrgUser.Permissions);
+        Assert.DoesNotContain("deleteAssignedCollections", migratedOrgUser.Permissions);
+    }
+
+    [DatabaseTheory, DatabaseData(MigrationName = _migrationName)]
+    public async Task RunMigration_WithoutAssignedCollectionsPermissions_DoesNothing(
+        IUserRepository userRepository,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IMigrationTesterService migrationTester)
+    {
+        // Setup data
+        var orgUser = await SetupData(
+            userRepository, organizationRepository, organizationUserRepository, OrganizationUserType.Custom,
+            editAssignedCollections: false, deleteAssignedCollections: false, accessEventLogs: false);
+        // Remove the editAssignedCollections and deleteAssignedCollections permissions
+        orgUser.Permissions = JsonSerializer.Serialize(new
+        {
+            AccessEventLogs = false,
+            AccessImportExport = false,
+            AccessReports = false,
+            CreateNewCollections = false,
+            EditAnyCollection = false,
+            DeleteAnyCollection = false,
+            ManageGroups = false,
+            ManagePolicies = false,
+            ManageSso = false,
+            ManageUsers = false,
+            ManageResetPassword = false,
+            ManageScim = false
+        }, JsonHelpers.CamelCase);
+        await organizationUserRepository.ReplaceAsync(orgUser);
+
+        // Run data migration
+        migrationTester.ApplyMigration();
+
+        // Assert that the user kept the Admin type and no changes were made to the permissions
+        var migratedOrgUser = await organizationUserRepository.GetByIdAsync(orgUser.Id);
+        Assert.NotNull(migratedOrgUser);
+        Assert.Equal(orgUser.Id, migratedOrgUser.Id);
+        Assert.Equal(orgUser.Type, migratedOrgUser.Type);
```

Comment:
Generally you should avoid comparing 2 mutable values like this. Both may have changed without us realising and this could pass. Better to break it out into separate assertions using a literal value, e.g.

```cs
        Assert.Equal(OrganizationUserType.Custom, orgUser.Type);
        Assert.Equal(OrganizationUserType.Custom, migratedOrgUser.Type);
```